lib/exceptions/backends/rollbar.rb

```diff
@@ -42,14 +42,32 @@ def clear_context
 
       private
 
+      # if reporting via the `error_class` and `error_message` args, we need to
+      # trick the rollbar reporter into thinking it's an actual exception. by
+      # doing that, rollbar will report the error class and message correctly,
+      # and show a stack trace, which it doesn't do when passing it a string.
+      def maybe_turn_into_exception(error_class_or_exception_or_string,
+                                    error_message)
+        if error_class_or_exception_or_string.is_a?(Exception)
+          error_class_or_exception_or_string
+        else
+          Class.new(StandardError) do
```

Comment:
hopefully these are garbage collected :)
(I believe they are)

Reply:
lol, good thing rollbar is accepting of these fakes. from my research, the only legit way of giving a class a name is by using `const_set`, which would've been crazy

Reply:
So... any chance these are not GC'd? We saw some OOMs start happening in r101-api when it was updated.

Reply:
```ruby
puts "pre"
puts "classes: #{ObjectSpace.count_objects[:T_CLASS]}"
puts "objects: #{ObjectSpace.count_objects[:T_OBJECT]}"
1_000_000.times do |i|
  Class.new(StandardError) do
    define_singleton_method(:name) do
      "name_#{i}"
    end

    define_method(:message) do
      "message_#{i}"
    end
  end.new
end
puts "post"
puts "classes: #{ObjectSpace.count_objects[:T_CLASS]}"
puts "objects: #{ObjectSpace.count_objects[:T_OBJECT]}"
GC.start
puts "post gc"
puts "classes: #{ObjectSpace.count_objects[:T_CLASS]}"
puts "objects: #{ObjectSpace.count_objects[:T_OBJECT]}"
```

```
pre
classes: 550
objects: 1687
post
classes: 3501
objects: 2576
post gc
classes: 552
objects: 1684
```

TL;DR they are